### PR TITLE
TOOLS-67569 Hadoop Partnership: Block Hadoop Client Teams from checki…

### PR DESCRIPTION
**Pull request by hungj**
…ng in Hadoop Jars in their MP

### CODE REVIEW

**Comment by convexquad** [comment]
Seconding comments from Anant. I think adding this to the DependencyPlugin that is a part of his PR is the right place for this functionality.

**Comment by nntnag17** [comment]
This looks pretty good to me. Once you hear from the tools team and address their comments, I'll be happy to +1 it.

File: hadoop-plugin/src/test/groovy/com/linkedin/gradle/dependency/DependencyPluginTest.groovy [hunk 1]
```diff
@@ -21,11 +21,15 @@ import org.gradle.api.Project;
...
 import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.artifacts.FileCollectionDependency;
+import org.gradle.api.artifacts.dsl.DependencyHandler;
 import org.gradle.api.internal.artifacts.dependencies.DefaultExternalModuleDependency;
+import org.gradle.api.internal.artifacts.dependencies.DefaultSelfResolvingDependency;
 import org.gradle.testfixtures.ProjectBuilder;
 import org.junit.Assert;
 import org.junit.Before;
 import org.junit.Test;
+import java.io.File;
```
Comment (convexquad) [inline hadoop-plugin/src/test/groovy/com/linkedin/gradle/dependency/DependencyPluginTest.groovy]: I don't think you need this import - java.io.File should be implicitly imported

**Comment by convexquad** [comment]
This looks much better - I'll let Anant drive the rest of the review.

**Comment by convexquad** [comment]
@hungj Looks like you need a rebase. Also can you bump the version number in gradle.properties, update VERSIONS.md and CONTRIBUTORS.md with your own info?

**Comment by hungj** [comment]
Hi @convexquad, @nntnag17,
I updated the PR to include comments from the gradle team. Right now the task looks in the project's build dir and verifies that all file dependencies are there - otherwise, it is a file dependency that should not be allowed.

**Comment by nntnag17** [comment]
Hi @hungj,

This looks pretty good. One minor comment, Could you squash your commits to a single commit. Apart from that +1 from my side.

**Pull request merged**